### PR TITLE
fates api 10

### PR DESCRIPTION
**Pull request by rgknox**
This set of changes modifies the interface with fates to accommodate API 10. 
This includes an updated fates parameter file. FATES API 10 focused on changes
to plant hydraulics and plant age.

[non-BFB]

### CODE REVIEW

**Comment by rgknox** [comment]
Ran regression tests against master, here is the FAIL log. 
```
[rgknox@n0003 rgknox]$ ./cs.status.E9f207e6-F0059286-api-10 | grep FAIL
    FAIL ERS.f09_g16.I1850CLM45CN.lawrencium-lr3_intel BASELINE Ea3be468-Fab1647a-master-v2: DIFF
    FAIL ERS.f19_f19.I1850CLM45CN.lawrencium-lr3_intel BASELINE Ea3be468-Fab1647a-master-v2: DIFF
    FAIL ERS.f19_g16.I1850CNRDCTCBC.lawrencium-lr3_intel.clm-rd BASELINE Ea3be468-Fab1647a-master-v2: DIFF
    FAIL ERS.f19_g16.I1850GSWCNPECACNTBC.lawrencium-lr3_intel.clm-eca_f19_g16_I1850GSWCNPECACNTBC BASELINE Ea3be468-Fab1647a-master-v2: DIFF
    FAIL ERS.f19_g16.I1850GSWCNPRDCTCBC.lawrencium-lr3_intel.clm-ctc_f19_g16_I1850GSWCNPRDCTCBC BASELINE Ea3be468-Fab1647a-master-v2: DIFF
    FAIL ERS.f19_g16.I20TRGSWCNPECACNTBC.lawrencium-lr3_intel.clm-eca_f19_g16_I20TRGSWCNPECACNTBC BASELINE Ea3be468-Fab1647a-master-v2: DIFF
  ERS_Ld3.f45_f45.ICLM45ED.lawrencium-lr3_intel.clm-fates (Overall: NLFAIL) details:
    FAIL ERS_Ld3.f45_f45.ICLM45ED.lawrencium-lr3_intel.clm-fates NLCOMP
  SMS.r05_r05.I1850CLM45CN.lawrencium-lr3_intel (Overall: FAIL) details:
    FAIL SMS.r05_r05.I1850CLM45CN.lawrencium-lr3_intel RUN time=102
    FAIL SMS_Ly2_P1x1.1x1_smallvilleIA.ICLM45CNCROP.lawrencium-lr3_intel.clm-lulcc_sville BASELINE Ea3be468-Fab1647a-master-v2: DIFF
    FAIL SMS_Ly2_P1x1.1x1_smallvilleIA.ICLM45CNCROP.lawrencium-lr3_intel.clm-lulcc_sville TPUTCOMP Error: Computation time increase > 25 pct from baseline
```
I'm not sure what to make of these BASELINE compare diffs.  I did check the cprnc output on the cpl files, and there are real differences in the output.  I can't make sense of it though, because these changes really shouldn't change non-fates behavior at all.  Anything obvious here?

**Comment by xuchongang** [comment]
@rgknox , I tested on the LANL badger machine.  Your last commit (9f207e69c0b4b43122643289fba142e7b3461f65) breaks the EDInit for plant hydraulics. Without this commit, the model runs without issue for BCI (one year). 

forrtl: severe (174): SIGSEGV, segmentation fault occurred
Image              PC                Routine            Line        Source             
e3sm.exe           00000000051B5E1D  Unknown               Unknown  Unknown
libpthread-2.17.s  00002B56229DA630  Unknown               Unknown  Unknown
e3sm.exe           00000000037CFB17  fatesplanthydraul         483  FatesPlantHydraulicsMod.F90
e3sm.exe           000000000390B583  edcohortdynamicsm         325  EDCohortDynamicsMod.F90
e3sm.exe           000000000365670E  fatesinventoryini        1131  FatesInventoryInitMod.F90
e3sm.exe           0000000003646980  fatesinventoryini         371  FatesInventoryInitMod.F90
e3sm.exe           00000000033498F8  edinitmod_mp_init         440  EDInitMod.F90
e3sm.exe           0000000000A6D938  clmfatesinterface        1357  clmfates_interfaceMod.F90
e3sm.exe           000000000090A542  clm_initializemod         910  clm_initializeMod.F90
e3sm.exe           0000000000883CC4  lnd_comp_mct_mp_l         273  lnd_comp_mct.F90
e3sm.exe           000000000045E48C  component_mod_mp_         257  component_mod.F90
e3sm.exe           0000000000423C8A  cime_comp_mod_mp_        1350  cime_comp_mod.F90
e3sm.exe           0000000000455215  MAIN__                    122  cime_driver.F90
e3sm.exe           000000000041146E  Unknown               Unknown  Unknown
libc-2.17.so       00002B562310F545  __libc_start_main     Unknown  Unknown
e3sm.exe           0000000000411369  Unknown               Unknown  Unknown

**Comment by rgknox** [comment]
ok @xuchongang , that last commit is merging in master, so there are some non-fates updates in there.  Also, can you confirm that you used the same version of FATES in both runs?

**Comment by rgknox** [comment]
@xuchongang and I are working on this offline, I think this will require some shuffling around of code on the FATES side.  This PR is not ready.

**Comment by rgknox** [comment]
merged up to master 86793b1

**Comment by rgknox** [comment]
Updated PR per issues @xuchongang and I found.  This necessitated some serious code refactoring on the FATES side, but I'm happy with the change in initialization sequence between the two models now.

Tested on lawrencium, tests should not b3 b4b with FATES, this should not change answers in non-fates tests.  My tests on lawrencium seem to always generate a lot of red-herrings, but I thought I would report them anyway.

```
rgknox@n0003 rgknox]$ ./cs.status.E56e5d1d-F7841aef-api11-v3 | grep FAIL
    FAIL ERS.f19_f19.I20TRCLM45CN.lawrencium-lr3_intel TPUTCOMP Error: Computation time increase > 25 pct from baseline
    FAIL ERS.f19_g16.I1850CLM45.lawrencium-lr3_intel.clm-betr BASELINE E86793b1-Fab1647a-master: DIFF
    FAIL ERS.f19_g16.I20TRGSWCNPECACNTBC.lawrencium-lr3_intel.clm-eca_f19_g16_I20TRGSWCNPECACNTBC BASELINE E86793b1-Fab1647a-master: DIFF
    FAIL ERS.f19_g16.I20TRGSWCNPRDCTCBC.lawrencium-lr3_intel.clm-ctc_f19_g16_I20TRGSWCNPRDCTCBC BASELINE E86793b1-Fab1647a-master: DIFF
    FAIL ERS.ne11_oQU240.I20TRCLM45.lawrencium-lr3_intel BASELINE E86793b1-Fab1647a-master: ERROR BFAIL some baseline files were missing
    FAIL ERS_Ld3.f45_f45.ICLM45ED.lawrencium-lr3_intel.clm-fates NLCOMP
    FAIL ERS_Ld3.f45_f45.ICLM45ED.lawrencium-lr3_intel.clm-fates BASELINE E86793b1-Fab1647a-master: DIFF
  SMS.r05_r05.I1850CLM45CN.lawrencium-lr3_intel (Overall: FAIL) details:
    FAIL SMS.r05_r05.I1850CLM45CN.lawrencium-lr3_intel RUN time=88
    FAIL SMS_Ld1.hcru_hcru.I1850CRUCLM45CN.lawrencium-lr3_intel TPUTCOMP Error: Computation time increase > 25 pct from baseline
```

Depending on team review, I think its ready for further testing.

**Comment by bishtgautam** [comment]
@rgknox, It is okay to rename the file in the future nutrient PR and I'm going to approve this PR. Let me know if you will be adding a new test and I will then merge this branch to next accordingly.

**Comment by rgknox** [comment]
I will add a new test, it actually caught a problem, stay tuned!

File: components/clm/src/main/clmfates_interfaceMod.F90 [hunk 1]
```diff
@@ -438,11 +438,19 @@ subroutine init(this, bounds_proc )
...
             ! INTERF-TODO: WE HAVE NOT FILTERED OUT FATES SITES ON INACTIVE COLUMNS.. YET
             ! NEED A RUN-TIME ROUTINE THAT CLEARS AND REWRITES THE SITE LIST
-
-            if (lun_pp%itype(l) == istsoil ) then
+            if ( lun_pp%itype(l) == istsoil ) then
                s = s + 1
                collist(s) = c
                this%f2hmap(nc)%hsites(c) = s
+
+!               if(.not.col_pp%active(c)) then
```
Comment (bishtgautam) [inline components/clm/src/main/clmfates_interfaceMod.F90:446]: @rgknox, How about deleting these commented out lines?

Comment (rgknox) [reply components/clm/src/main/clmfates_interfaceMod.F90:446]: doh, sorry, I should had caught that, thanks

**Comment by rgknox** [comment]
fyi, I am investigating the results of the new fates-hydro test.  Currently, it does not pass.  It is not clear yet if the issue has more to do with the solver or things on the FATES side (irrelevant to this PR), or if this is a coupling issue (relevant to this PR).

**Comment by bishtgautam** [comment]
@rgknox Are you still working on this PR or is it ready for merging?

**Comment by rgknox** [comment]
@bishtgautam , lets move ahead.  The new ERS_Ld5.f45_f45.ICLM45ED.clm-fateshydro test should be flagged as an expected fail for now.  Figuring out what is going on there is in my personal queue.

Note, I also extended the regular FATES test to 20 days.  If this is a time bottleneck compared to other nightly tests, its fine to shorten the test.

**Comment by bishtgautam** [comment]
This is now on next (https://github.com/E3SM-Project/E3SM/commit/c52d3ab828af57ec90e33134ffc2297c961031bd). While merging I forgot to add `#` in the commit message, so the reference to the merge commit doesn't show up here.

**Comment by dqwu** [comment]
@bishtgautam 
I tested latest next branch on anlworkstation and the following test failed:
ERS_Ld5.f45_f45.ICLM45ED.anlworkstation_gnu.clm-fateshydro

Based on the output log, the failure is likely caused by this PR.
```
[2]  ENDRUN:ERROR in /sandbox/jenkins-ci/workspace/E3SM_SCORPIO/E3SM_SCORPIO/components/clm/src/external_models/fates/biogeophys/FatesPlantHydraulicsMod.F90 at line 3231                                                                                                                                                                                                                                                                                                                                                                   
[2]  ERROR: Unknown error submitted to shr_abort_abort.
[2] #0  0xb9ebda in ???
[2] #1  0xb9ed9d in ???
[2] #2  0x951598 in ???
[2] #3  0x9587a1 in ???
[2] #4  0x4e8494 in ???
[2] #5  0x6ccc4d in ???
[2] #6  0x4bee5f in ???
[2] #7  0x4b6446 in ???
[2] #8  0x423d90 in ???
[2] #9  0x412081 in ???
[2] #10  0x409727 in ???
[2] #11  0x7f4750ca8f44 in ???
[2] #12  0x409851 in ???
[2] #13  0xffffffffffffffff in ???
[2] application called MPI_Abort(MPI_COMM_WORLD, 1001) - process 2
```

**Comment by rgknox** [comment]
@dqwu, this is an expected fail, and is a new test as of this PR.
Should I remove the test from this PR until I can verify the test passes?

**Comment by rljacob** [comment]
Yes remove the test until it is working.

**Comment by rgknox** [comment]
test was removed

**Pull request merged**